File: Algorithms/LeetCode_1089.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        i=0
        while i != len(arr)-1:
            
            if(arr[i] == 0):
                index=i+1
                for j in range(len(arr)-1,index,-1):
                    arr[j]=arr[j-1]
                arr[index]=0
                i+=1    
            if i == len(arr)-1:
                break
            i+=1        
                    

File: Algorithms/LeetCode_1089.java
class Solution {
    public void duplicateZeros(int[] arr) {
        for(int i=0;i<arr.length;i++){
            if(i == arr.length-1) break;
            if(arr[i] == 0){
                int index=i+1;
                for(int j=arr.length-1;j>index;j--){
                    arr[j]=arr[j-1];
                }
                arr[index]=0;
                i++;
            }
        }
    }
}
